modules/stochastic_tools/include/utils/POD.h

```diff
@@ -0,0 +1,58 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ParallelSolutionStorage.h"
+#include <slepcsvd.h>
+#include "libmesh/petsc_vector.h"
+#include <MooseTypes.h>
+/**
+ * Class which computes a Proper Orthogonal Decomposition (POD) for snapshots
+ * stored in ParallelSolutionStorage.
+ */
+class POD
+{
+public:
+  POD(const ParallelSolutionStorage * const _parallel_storage,
+      const std::string & _extra_slepc_options,
+      const Parallel::Communicator & communicator);
+  /**
+   *
+   * @param vname Variable name to extract snapshot data
+   * @param left_basis_functions Vector for left basis function
+   * @param right_basis_functions Vector for right basis functions
+   * @param singular_values Vector for singular values
+   * @param num_modes Max number of modes to compute
+   * @param energy Energy threshold to determine the minimum number of modes
+   */
+  void computePOD(const VariableName & vname,
+                  std::vector<DenseVector<Real>> & left_basis_functions,
+                  std::vector<DenseVector<Real>> & right_basis_functions,
+                  std::vector<Real> & singular_values,
+                  const dof_id_type & num_modes,
+                  const Real & energy) const;
```

Comment:
```suggestion
                  const Real energy) const;
```

Reply:
In terms of memory a Real and a reference to a Real are the same so in this case it is not useful to pass as a reference. At the same time, this is more of a style-based comment so feel free to disregard. Especially that compared to the other operations in this function this might be a drop in the ocean.

Reply:
If you choose to add this, please modify the function below as well.